test-data/index.ts

```diff
@@ -0,0 +1,391 @@
+            value: 'new value 3',
+            occurrences: 7
+          },
+          {
+            xPath: '//rdf:RDF/edm:ProvidedCHO/@rdf:about',
+            value: 'new value 4',
+            occurrences: 6
+          }
+        ]
+      }
+    ]
+  });
+}
+
+function getStatistics(): string {
+  return JSON.stringify({
+    taskId: -5886403149040825151,
+    nodePathStatistics: [
+      {
+        xPath: '//rdf:RDF/edm:ProvidedCHO',
+
+        nodeValueStatistics: [
+          {
+            value: 'originally loaded A 1',
+            occurrences: 818,
+            attributeStatistics: [
+              {
+                xPath: '//rdf:RDF/edm:ProvidedCHO/@rdf:about',
+                value: 'orig attr stat value 1-1',
+                occurrences: 12
+              },
+              {
+                xPath: '//rdf:RDF/edm:ProvidedCHO/@rdf:about',
+                value: 'orig attr stat value 1-2',
+                occurrences: 21
+              }
+            ]
+          },
+          {
+            value: 'originally loaded A 2',
+            occurrences: 919,
+            attributeStatistics: [
+              {
+                xPath: '//rdf:RDF/edm:ProvidedCHO/@rdf:about',
+                value: 'orig attr stat value 1-3',
+                occurrences: 12
+              },
+              {
+                xPath: '//rdf:RDF/edm:ProvidedCHO/@rdf:about',
+                value: 'orig attr stat value 1-4',
+                occurrences: 21
+              }
+            ]
+          }
+        ]
+      },
+      {
+        xPath: '//rdf:RDF/edm:Agent',
+
+        nodeValueStatistics: [
+          {
+            value: 'originally loaded B 1',
+            occurrences: 11111111,
+
+            attributeStatistics: [
+              {
+                xPath: '//rdf:RDF/edm:Agent/@rdf:about',
+                value: 'orig attr stat value 2-1',
+                occurrences: 46
+              },
+              {
+                xPath: '//rdf:RDF/edm:Agent/@ns2:about',
+                value: 'orig attr stat value 2-2',
+                occurrences: 42
+              }
+            ]
+          },
+          {
+            value: 'originally loaded B 2',
+            occurrences: 22222222,
+
+            attributeStatistics: [
+              {
+                xPath: '//rdf:RDF/edm:Agent/@rdf:about',
+                value: 'orig attr stat value 2-3',
+                occurrences: 46
+              },
+              {
+                xPath: '//rdf:RDF/edm:Agent/@ns2:about',
+                value: 'orig attr stat value 2-4',
+                occurrences: 42
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  });
+}
+
+function routeToFile(response: ServerResponse, route: string): boolean {
+  response.setHeader('Content-Type', 'application/json;charset=UTF-8');
+
+  let regRes = route.match(/orchestrator\/proxies\/(\D+)\/task\/-?(\d+)\/report\/exists/);
+
+  if (regRes) {
```

Comment:
I am not sure, but can this route.match - and the ones further down - go directly in the if? 

Eg: 
`if (route.match(/orchestrator\/proxies\/(\D+)\/task\/-?(\d+)\/report\/exists/)) { .... }`

Reply:
I have to access the match in the return of the function.  It can be accessed (but not declared) withing the if, so something like this is possible:

let x;
if(x=1 > 0){ console.log('yes...'); }

but is this desirable?  It's discouraged, allegedly, within Java...

Reply:
Thanks, that makes sense, no need to change.